src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+      case 'clear' :
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
+    }
+
+    const assign = { ...stateCopy };
+
+    stateHistory.push(assign);
```

Comment:
You don't need to create a new variable. Just use `{ ...stateCopy }` in `push`
```suggestion

    stateHistory.push( { ...stateCopy });
```